web/datapipeline/sap_systems_projector.go

```diff
@@ -0,0 +1,84 @@
+package datapipeline
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/trento/internal/sapsystem"
+	"github.com/trento-project/trento/web/entities"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+func NewSAPSystemsProjector(db *gorm.DB) *projector {
+	SAPSystemsProjector := NewProjector("sapsystems", db)
+
+	SAPSystemsProjector.AddHandler(SAPsystemDiscovery, SAPSystemsProjector_SAPSystemsDiscoveryHandler)
+
+	return SAPSystemsProjector
+}
+
+func SAPSystemsProjector_SAPSystemsDiscoveryHandler(dataCollectedEvent *DataCollectedEvent, db *gorm.DB) error {
+	decoder := getPayloadDecoder(dataCollectedEvent.Payload)
+	var discoveredSAPSystems sapsystem.SAPSystemsList
+	if err := decoder.Decode(&discoveredSAPSystems); err != nil {
+		log.Errorf("can't decode data: %s", err)
+		return err
+	}
+
+	for _, s := range discoveredSAPSystems {
+		instance := entities.SAPSystemInstance{
+			AgentID:  dataCollectedEvent.AgentID,
+			SystemID: s.Id,
```

Comment:
I'm a bit troubled with the name, but I recognize that it is not simple hehe
Maybe, we could rename `SystemID` to simply `ID`? `SID` at the end is `system id` as well
It is the internal ID of the system at the end.
By the way, in the future we will need to convert this to `uuid` format. Nothing to be done in the this PR though

Reply:
@arbulu89 I asked @dottorblaster  to change this 🍡 
To be aligned, I see this expressed as SystemID in the UI, what is the SAP domain word for this one?

Reply:
Tough decision but `ID` is a slippery call :sweat_smile: I would prefer having little longer field names, and `ID` is just... well an ID? :smile:

Reply:
The thing is that `SystemID` and `SID` are the same as `SID` stands for `System identifier`. The thing is that our `SystemID` is a an internal auto-generated ID that is **global**. The sap `SID` never was created to be global, that's why it is just a 3 chars/digits string.
In this case the `ID` is our ID, the identifier of the SAP system (not the instance).
Up to you guys, I just wanted to comment that `SystemID` and `SID` are synonims hehe One is used by SAP and the other by us

Reply:
Ah alright i thought that `SystemID` was our `ID` and `SID` was another thing :D

Reply:
Yes, and it is.
`SystemID` our internal ID
`SID` sap system id -> `System identifier`
The difference is that one is ours and the other SAP internal hehe